core/lib/types/src/fee_model.rs

```diff
@@ -0,0 +1,185 @@
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct L1PeggedBatchFeeModelInput {
+    /// Fair L2 gas price to provide
+    pub fair_l2_gas_price: u64,
+    /// The L1 gas price to provide to the VM.
+    pub l1_gas_price: u64,
+}
+
+/// Pubdata price may be independent from L1 gas price.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct PubdataIndependentBatchFeeModelInput {
+    /// Fair L2 gas price to provide
+    pub fair_l2_gas_price: u64,
+    /// Fair pubdata price to provide.
+    pub fair_pubdata_price: u64,
+    /// The L1 gas price to provide to the VM. Even if some of the VM versions may not use this value, it is still maintained for backward compatibility.
+    pub l1_gas_price: u64,
+}
+
+/// The enum which represents the version of the fee model. It is used to determine which fee model should be used for the batch.
+/// - `V1`, the first model that was used in zkSync Era. In this fee model, the pubdata price must be pegged to the L1 gas price.
+/// Also, the fair L2 gas price is expected to only include the proving/computation price for the operator and not the costs that come from
+/// processing the batch on L1.
+/// - `V2`, the second model that was used in zkSync Era. There the pubdata price might be independent from the L1 gas price. Also,
+/// The fair L2 gas price is expected to both the proving/computation price for the operator and the costs that come from
+/// processing the batch on L1.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum MainNodeFeeModelConfig {
+    V1(MainNodeFeeModelConfigV1),
+    V2(MainNodeFeeModelConfigV2),
+}
+
+/// Config params for the first version of the fee model. Here, the pubdata price is pegged to the L1 gas price and
+/// neither fair L2 gas price nor the pubdata price include the overhead for closing the batch
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct MainNodeFeeModelConfigV1 {
+    /// The minimal acceptable L2 gas price, i.e. the price that should include the cost of computation/proving as well
+    /// as potentially premium for congestion.
+    /// Unlike the `V2`, this price will be directly used as the `fair_l2_gas_price` in the bootloader.
+    pub minimal_l2_gas_price: u64,
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct MainNodeFeeModelConfigV2 {
+    /// The minimal acceptable L2 gas price, i.e. the price that should include the cost of computation/proving as well
+    /// as potentially premium for congestion.
+    pub minimal_l2_gas_price: u64,
+    /// The constant that represents the possibility that a batch can be sealed because of overuse of computation resources.
+    /// It has range from 0 to 1. If it is 0, the compute will not depend on the cost for closing the batch.
+    /// If it is 1, the gas limit per batch will have to cover the entire cost of closing the batch.
+    pub compute_overhead_percent: f64,
```

Comment:
btw 'percent' in the name suggests that it is in the range from 0 to 100...

maybe call it 'batch_sealed_due_to_compute_odds' ? 
(and then the other one would be 'batch_sealed_due_to_pubdata_odds' )

Reply:
maybe better to call it `part`, since it basically defines which part of the batch overhead will the compute take?

Reply:
IMHO `odds` sounds a bit weird in this context (as it is not really a probability, but rather a percentage of the batch overhead to be paid by compute)

Reply:
we could also say 'probability' instead of odds..

so it would be:
'batch_sealed_due_to_compute_probability'